tempesta_fw/http.c

```diff
@@ -2930,6 +2934,37 @@ tfw_http_req_process(TfwConn *conn, const TfwFsmData *data)
 		skb = NULL;
 	}
 
+	/*
+	 * If a client work through a forward proxy, then a proxy can pass it's
+	 * IP address by the first value in X-Forwarded-For
+	 */
+	s_xff = req->h_tbl->tbl[TFW_HTTP_HDR_X_FORWARDED_FOR];
+	s_ip = tfw_str_next_str_val(&s_xff);
+	nchunks = 0;
+	TFW_STR_FOR_EACH_CHUNK(c, &s_ip, end) {
+		if (!(c->flags & TFW_STR_VALUE))
+			break;
+		nchunks++;
+	}
+	s_ip.nchunks = nchunks;
+
+	if (TFW_STR_EMPTY(&s_ip) || tfw_addr_pton(&s_ip, &addr) != 0) {
+		ss_getpeername(conn->sk, &addr);
+	}
```

Comment:
There is no sense to execute the code if we have no User-Agent since the client was already obtained in `tfw_sock_clnt_new()` for the same address.

BTW we do not use braces for single-line IF statements unless there is multi-line ELSE.